### ISSUE TITLE
source links not working as expected

### ISSUE THREAD

**Issue by oguiza** [fastai/nbdev#468]
Hi,

First of all thanks a lot for creating this great library!

I've used nbdev to build a time series library on top of fastai & Pytorch (https://github.com/timeseriesAI/tsai).

I've just noticed that the links to [source] in the documentation don't work. Here's an example. The link points to: 
```
https://timeseriesai.github.io/tsai/tsai/data/core.py#L20 
```
when it should point to:
```
https://github.com/timeseriesAI/tsai/tree/main/tsai/data/core.py#L20
```

I've set my settings based on nbdev's settings: 
```
[DEFAULT]
host = github
lib_name = tsai
user = timeseriesAI
branch = main
version = 0.2.18
description = Practical Deep Learning for Time Series / Sequential Data library based on fastai & Pytorch
keywords = fastai time-series time-series-classification time-series-regression deep-learning Pytorch
author = Ignacio Oguiza and contributors
author_email = timeseriesAI@gmail.com
title = tsai
copyright = Ignacio Oguiza
license = apache2
status = 3
min_python = 3.6
audience = Developers
language = English
requirements = fastai>=2.3.0 scikit-learn>=0.24.1 scipy>=1.5 sktime>=0.5.3 statsmodels>=0.12.2 pyunpack>=0.2.2 PyWavelets>=1.1.1 numba>=0.53.1 pyts>=0.11.0 imbalanced-learn>=0.8.0 dask>=2021.3.0 argparse>=1.4.0 tsfresh>=0.18.0 torch-optimizer>=0.1.0
pip_requirements = torch>=1.7.0,<1.8
conda_requirements = pytorch>=1.7.0,<1.8
console_scripts = nbdev_build_lib=nbdev.export2html:nbdev_build_lib
	nbdev_update_lib=nbdev.sync:nbdev_update_lib
	nbdev_diff_nbs=nbdev.sync:nbdev_diff_nbs
	nbdev_test_nbs=nbdev.test:nbdev_test_nbs
	nbdev_build_docs=nbdev.export2html:nbdev_build_docs
	nbdev_nb2md=nbdev.export2html:nbdev_nb2md
	nbdev_trust_nbs=nbdev.sync:nbdev_trust_nbs
	nbdev_clean_nbs=nbdev.clean:nbdev_clean_nbs
	nbdev_read_nbs=nbdev.test:nbdev_read_nbs
	nbdev_fix_merge=nbdev.merge:nbdev_fix_merge
	nbdev_install_git_hooks=nbdev.cli:nbdev_install_git_hooks
	nbdev_bump_version=nbdev.cli:nbdev_bump_version
	nbdev_new=nbdev.cli:nbdev_new
	nbdev_detach=nbdev.export2html:nbdev_detach
nbs_path = nbs
doc_path = docs
doc_host = https://timeseriesAI.github.io
doc_baseurl = /
git_url = https://github.com/timeseriesAI/tsai/tree/main/
lib_path = tsai
custom_sidebar = True
recursive = True
cell_spacing = 1
monospace_docstrings = False
```
I don't know if I'm doing something wrong, but have reviewed the documentation, forum, and issues and have not been able to find a solution. 
Thanks a lot in advance for your help!

**Comment by oguiza** [comment]
I've noticed that recursive is set to True in nbdev. I've changed it from False to True but it didn't make any difference. source links still don't work.

**Comment by oguiza** [comment]
In the end I managed to resolve this in a very manual way. Building a new library and copying over the files. Lot's of work, but it's working well now.

**Comment by caseytomlin** [comment]
Hi @oguiza, I'm not entirely sure it's related since I am using nbdev on gitlab, but I faced a similar issue when I wanted to change the branch from which the pages documentation is generated. For me, after making the branch change in `settings.ini` and `.gitlab-ci.yml`, the source links did not update in pages, which I think may be due to the fact that `write_tmpl` (https://github.com/fastai/nbdev/blob/master/nbdev/export2html.py#L440) will not update the `docs/_config.yml` and `docs/_data/topnav.yml` if they exist already, so my solution was to delete them and rerun `nbdev_build_lib`. 

adding @tcapelle in case this is somehow gitlab-specific (in my case I also had to add a `recursive` variable to `settings.ini` which is not in the @tcapelle's template to avoid a `write_tmpl` key error during the rebuild).
